crates/ide_assists/src/handlers/inline_type_alias.rs

```diff
@@ -0,0 +1,714 @@
+    AssistId, AssistKind,
+};
+
+// Assist: inline_type_alias
+//
+// Replace a type alias with its concrete type.
+//
+// ```
+// type A<T = u32> = Vec<T>;
+//
+// fn main() {
+//     let a: $0A;
+// }
+// ```
+// ->
+// ```
+// type A<T = u32> = Vec<T>;
+//
+// fn main() {
+//     let a: Vec<u32>;
+// }
+// ```
+pub(crate) fn inline_type_alias(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let alias_instance = ctx.find_node_at_offset::<ast::PathType>()?;
+    let alias = get_type_alias(&ctx, &alias_instance)?;
+    let concrete_type = alias.ty()?;
+
+    let replacement = if let Some(alias_generics) = alias.generic_param_list() {
+        get_replacement_for_generic_alias(
+            alias_instance.syntax().descendants().find_map(ast::GenericArgList::cast),
+            alias_generics,
+            &concrete_type,
+        )?
+    } else {
+        concrete_type.to_string()
+    };
+
+    let target = alias_instance.syntax().text_range();
+
+    acc.add(
+        AssistId("inline_type_alias", AssistKind::RefactorInline),
+        "Inline type alias",
+        target,
+        |builder| {
+            builder.replace(target, replacement);
+        },
+    )
+}
+
+/// This doesn't attempt to ensure specified generics are compatible with those
+/// required by the type alias, other than lifetimes which must either all be
+/// specified or all omitted. It will replace TypeArgs with ConstArgs and vice
+/// versa if they're in the wrong position. It supports partially specified
+/// generics.
+///
+/// 1. Map the provided instance's generic args to the type alias's generic
+///    params:
+///
+///    ```
+///    type A<'a, const N: usize, T = u64> = &'a [T; N];
+///          ^ alias generic params
+///    let a: A<100>;
+///            ^ instance generic args
+///    ```
+///
+///    generic['a] = '_ due to omission
+///    generic[N] = 100 due to the instance arg
+///    generic[T] = u64 due to the default param
+///
+/// 2. Copy the concrete type and substitute in each found mapping:
+///
+///    &'_ [u64; 100]
+///
+/// 3. Remove wildcard lifetimes entirely:
+///
+///    &[u64; 100]
+fn get_replacement_for_generic_alias(
+    instance_generic_args_list: Option<ast::GenericArgList>,
+    alias_generics: ast::GenericParamList,
+    concrete_type: &ast::Type,
+) -> Option<String> {
+    if alias_generics.generic_params().count() == 0 {
+        cov_mark::hit!(no_generics_params);
+        return None;
+    }
+
+    let mut lifetime_mappings = HashMap::<&str, ast::Lifetime>::new();
+    let mut other_mappings = HashMap::<String, SyntaxNode>::new();
+
+    let wildcard_lifetime = make::lifetime("'_");
+    let alias_lifetimes = alias_generics.lifetime_params().map(|l| l.to_string()).collect_vec();
+    for lifetime in &alias_lifetimes {
+        lifetime_mappings.insert(lifetime, wildcard_lifetime.clone());
+    }
+
+    if let Some(ref instance_generic_args_list) = instance_generic_args_list {
```

Comment:
```suggestion
    if let Some(instance_generic_args_list) = &instance_generic_args_list {
```
While I do love `ref` unlike most people, there is no need to use it here :)